packages/truffle/test/scenarios/commandRunner.js

```diff
@@ -60,11 +60,16 @@ module.exports = {
         ? `debug(${displayHost})>`
         : `truffle(${displayHost})>`;
 
-    let seenChildPrompt = false;
+    // seems safe to escape parens only, as the readyprompt is constructed from
+    // [a-zA-Z] strings and wrapping parens.
+    const escapedPrompt = readyPrompt.replace("(", "\\(").replace(")", "\\)");
+    const readyPromptRex = new RegExp(`^${escapedPrompt}`, "gm");
```

Comment:
Sorry @cds-amal if it is a silly question, but what is this `"gm"` flag does? `"g"` is for global search and `"m"` is multi-line search if I am understanding correctly.

Reply:
Yes, "g" is for global. It means the meta characters, `^`, and `$` will match the beginning and end of each line instead of the beginning and end of the input string (which can contain multiple lines)